src/User/AvatarValidator.php

```diff
@@ -66,7 +66,11 @@ protected function assertFileMimes(UploadedFileInterface $file)
 
         $guessedExtension = MimeTypes::getDefault()->getExtensions($file->getClientMediaType())[0] ?? null;
 
-        if (! in_array($guessedExtension, $allowedTypes)) {
+        $stream = $file->getStream();
+        $tmpFilename = $stream->getMetadata('uri');
+        $notImage = ! getimagesize($tmpFilename);
+
+        if ($notImage || ! in_array($guessedExtension, $allowedTypes)) {
```

Comment:
`getimagesize` doesn't seem to be a reliable way of checking if the file is an image (https://www.php.net/manual/en/function.getimagesize.php)

I'm thinking it might be better to catch errors when creation the intervention image object, logging the error in the logs, and returning a better response to the end user.

Reply:
Changed to use that logic

Reply:
I wouldn't get rid of the mimes check though 🤔 ? the try catch should act as a second check.